{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "eacfa1b2-4ca0-413e-99e8-8d265e385c6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import gensim\n",
    "from gensim import utils\n",
    "import numpy as np\n",
    "import sys\n",
    "from sklearn.datasets import fetch_20newsgroups\n",
    "from nltk import word_tokenize\n",
    "from nltk import download\n",
    "from nltk.corpus import stopwords\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.decomposition import PCA\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f05633f6-8ae0-4e21-9530-156081c6662b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package stopwords to\n",
      "[nltk_data]     /Users/sunilmcesh/nltk_data...\n",
      "[nltk_data]   Package stopwords is already up-to-date!\n",
      "[nltk_data] Downloading package punkt to\n",
      "[nltk_data]     /Users/sunilmcesh/nltk_data...\n",
      "[nltk_data]   Package punkt is already up-to-date!\n",
      "[nltk_data] Downloading package wordnet to\n",
      "[nltk_data]     /Users/sunilmcesh/nltk_data...\n",
      "[nltk_data]   Package wordnet is already up-to-date!\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import glob\n",
    "import string\n",
    "import glob\n",
    "from tqdm import tqdm \n",
    "#import pdfminer\n",
    "from pdfminer.high_level import extract_text\n",
    "import pandas as pd\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "import nltk \n",
    "nltk.download('stopwords')\n",
    "import numpy as np\n",
    "from sklearn.pipeline import Pipeline\n",
    "from sklearn.feature_extraction.text import TfidfTransformer\n",
    "from sklearn.feature_extraction.text import CountVectorizer\n",
    "import nltk\n",
    "nltk.download('punkt')\n",
    "from nltk import sent_tokenize\n",
    "from nltk import word_tokenize\n",
    "from nltk.corpus import stopwords\n",
    "from nltk.stem import WordNetLemmatizer\n",
    "nltk.download('wordnet')\n",
    "from nltk.probability import FreqDist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "af8bbd1a-b0bf-4980-b8c6-2bdb849e06a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "from termcolor import colored"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b381dc7d-e1e7-41af-9058-c9e24dbcc60c",
   "metadata": {},
   "outputs": [],
   "source": [
    "PAPER_OF_INTEREST_FNAME  = glob.glob('/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/*/*/*.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "913606bf-698f-4407-ba3f-76b5d88b44aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Currently doesn't understand multiple insitutions for the first author\n",
    "# Doesn't remove if the first author has > 3 names \n",
    "def Clean_PDF(element):\n",
    "    score = 0 \n",
    "    # Remove DOI\n",
    "    X = re.search('^doi', element)\n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    # Remove Summary\n",
    "    X = re.search('^summary',  element)\n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    # Remove Software\n",
    "    X = re.search('^software', element)\n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    # Remove Review Repository Archive\n",
    "    X = re.search('^• review', element)\n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    X = re.search('• repository', element)\n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    X = re.search('• archive', element)\n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    # Submitted - Published\n",
    "    X = re.search('^submitted', element)\n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    X = re.search('^published', element)\n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    # Remove Licence CC-BY\n",
    "    X = re.search('^license', element)\n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    X = re.search('^licence', element)\n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    # Remove Page Number\n",
    "    if len(element.split(' ')) == 1:\n",
    "        score += 1\n",
    "    # Remove footer\n",
    "    X = re.search('et al.,', element)\n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    # Remove Author Names\n",
    "    if len(element.split(' ')) >1:\n",
    "        if element.split(' ')[1][-1] == '1':\n",
    "            score += 1\n",
    "        if element.split(' ')[1][-2:] == '1,':\n",
    "            score += 1\n",
    "        if element.split(' ')[1][-2:] == '1\\n':\n",
    "            score += 1\n",
    "    \n",
    "    if len(element.split(' ')) >2:\n",
    "        if element.split(' ')[2][-1] == '1':\n",
    "            score += 1\n",
    "        if element.split(' ')[2][-2:] == '1\\n':\n",
    "            score += 1\n",
    "    # Remove Insitution Names     \n",
    "    if element[0] == '1':\n",
    "        score += 1\n",
    "        if element[1] =='.':\n",
    "            score = score - 1 # Attempting to include bullet points\n",
    "    # Remove Future Woeks\n",
    "    X = re.search('^future work', element)\n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    # Remove references\n",
    "    X = re.search('^references', element) # Reference Title\n",
    "    if element.split(' ')[0][-1] == ',':\n",
    "        score += 1\n",
    "    if element.split(' ')[0][-1] == '.':\n",
    "        score += 1\n",
    "        if len(element.split(' ')[0]) <3:\n",
    "            score = score - 1 # Attempting to include bullet points\n",
    "    if len(element.split(' ')) >2:\n",
    "        if element.split(' ')[2][-1] == '.':\n",
    "            score += 1\n",
    "    # Remove Figures\n",
    "    X = re.search('^figure', element) \n",
    "    temp = X != None\n",
    "    score += temp\n",
    "    \n",
    "    return score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "4c60bfbb-9e31-4250-80a0-147eeebccad3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00027/10.21105.joss.00027.pdf\n"
     ]
    }
   ],
   "source": [
    "K = 11\n",
    "print(PAPER_OF_INTEREST_FNAME[K])\n",
    "Paper_interest = PAPER_OF_INTEREST_FNAME[K]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "dc4238ad-be24-4dba-b167-bd3b2258d859",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32msourmash: a library for minhash sketching of dna\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00027\n",
      "\u001b[0m\n",
      "\u001b[31m1 university of california, davis\n",
      "\u001b[0m\n",
      "\u001b[31mc. titus brown1 and luiz irber1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32msourmash is a toolbox for creating, comparing, and manipulating minhash sketches of\n",
      "genomic data.\n",
      "\u001b[0m\n",
      "\u001b[32mminhash sketches provide a lightweight way to store “signatures” of large dna or rna\n",
      "sequence collections, and then compare or search them using a jaccard index. minhash\n",
      "sketches can be used to identify samples, find similar samples, identify data sets with\n",
      "shared sequences, and build phylogenetic trees (ondov et al. 2015).\n",
      "\u001b[0m\n",
      "\u001b[32msourmash provides a command line script, a python library, and a cpython module for\n",
      "minhash sketches.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mondov, brian d, todd j treangen, adam b mallonee, nicholas h bergman, sergey koren,\n",
      "and adam m phillippy. 2015. “fast genome and metagenome distance estimation using\n",
      "minhash.” biorxiv. cold spring harbor labs journals, 029827. doi:10.1101/029827.\n",
      "\u001b[0m\n",
      "\u001b[31mbrown et al., (2016). sourmash: a library for minhash sketching of dna. journal of open source software, 1(5), 27, doi:10.21105/joss.00027 1\n",
      "\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "texts = ''\n",
    "arr = []\n",
    "from pdfminer.high_level import extract_pages\n",
    "from pdfminer.layout import LTTextContainer\n",
    "for page_layout in extract_pages(Paper_interest):\n",
    "    for element in page_layout:\n",
    "        if isinstance(element, LTTextContainer):\n",
    "            score = Clean_PDF(element.get_text().lower())\n",
    "            #print(score)\n",
    "            if score == 0:\n",
    "                print(colored(element.get_text().lower(), 'green'))\n",
    "            else:\n",
    "                print(colored(element.get_text().lower(), 'red'))\n",
    "                #arr.append(element.get_text())\n",
    "            #    texts = texts + ' ' + element.get_text()\n",
    "arr = np.array(arr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "3ffafd9e-fd22-462f-9dc9-7010694c8858",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Show_Cleaned_text(Paper_interest):\n",
    "    for page_layout in extract_pages(Paper_interest):\n",
    "        for element in page_layout:\n",
    "            if isinstance(element, LTTextContainer):\n",
    "                score = Clean_PDF(element.get_text().lower())\n",
    "            #print(score)\n",
    "                if score == 0:\n",
    "                    print(colored(element.get_text().lower(), 'green'))\n",
    "                else:\n",
    "                    print(colored(element.get_text().lower(), 'red'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "55dda8e7-dd90-43a7-874d-9d9687737b8f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00011/10.21105.joss.00011.pdf\n",
      "\u001b[32mcarl: a likelihood-free inference toolbox\n",
      "\u001b[0m\n",
      "\u001b[31mgilles louppe1, kyle cranmer1, and juan pavez2\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00011\n",
      "\u001b[0m\n",
      "\u001b[31m1 new york university 2 federico santa maría university\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mcarl is a toolbox for likelihood-free inference in python.\n",
      "\u001b[0m\n",
      "\u001b[32mthe likelihood function is the central object that summarizes the information from an\n",
      "experiment needed for inference of model parameters. it is key to many areas of science\n",
      "that report the results of classical hypothesis tests or confidence intervals using the (gen-\n",
      "eralized or profile) likelihood ratio as a test statistic. at the same time, with the advance\n",
      "of computing technology, it has become increasingly common that a simulator (or gener-\n",
      "ative model) is used to describe complex processes that tie parameters of an underlying\n",
      "theory and measurement apparatus to high-dimensional observations. however, directly\n",
      "evaluating the likelihood function in these cases is often impossible or is computationally\n",
      "impractical.\n",
      "\u001b[0m\n",
      "\u001b[32min this context, the goal of this package is to provide tools for the likelihood-free setup,\n",
      "including likelihood (or density) ratio estimation algorithms, along with helpers to carry\n",
      "out inference on top of these.\n",
      "\u001b[0m\n",
      "\u001b[32mapproximating likelihood ratios with calibrated classifiers\n",
      "\u001b[0m\n",
      "\u001b[32mmethodological details regarding likelihood-free inference with calibrated classifiers can\n",
      "be found in the companion paper (cranmer, pavez, and louppe 2016).\n",
      "\u001b[0m\n",
      "\u001b[32mfuture development aims at providing further density ratio estimation algorithms, along\n",
      "with alternative algorithms for the likelihood-free setup, such as approximate bayesian\n",
      "computation (abc).\n",
      "\u001b[0m\n",
      "\u001b[31mfuture works\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mcranmer, kyle, juan pavez, and gilles louppe. 2016. “approximating likelihood ratios\n",
      "with calibrated discriminative classifiers,” march. http://arxiv.org/abs/1506.02169v2.\n",
      "\u001b[0m\n",
      "\u001b[31mlouppe et al., (2016). carl: a likelihood-free inference toolbox. journal of open source software, 1(1), 11, doi:10.21105/joss.00011\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00012/10.21105.joss.00012.pdf\n",
      "\u001b[32mmst_clustering: clustering via euclidean minimum\n",
      "spanning trees\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00012\n",
      "\u001b[0m\n",
      "\u001b[31m1 university of washington escience institute\n",
      "\u001b[0m\n",
      "\u001b[31mjake vanderplas1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mthis package contains a python implementation of a clustering algorithm based on\n",
      "an efficiently-constructed approximate euclidean minimum spanning tree (described in\n",
      "(ivezić et al. 2014)). the method produces a hierarchical clustering of input data, and is\n",
      "quite similar to single-linkage agglomerative clustering. the advantage of this implemen-\n",
      "tation is the ability to find significant clusters even in the presence of background noise,\n",
      "and is particularly useful for researchers hoping to detect structure in physical data.\n",
      "\u001b[0m\n",
      "\u001b[32mthe code makes use of tools within scipy (jones et al. 2001–2001--) and scikit-learn (pe-\n",
      "dregosa et al. 2011), and is designed for compatibility with the scikit-learn api (buitinck\n",
      "et al. 2013).\n",
      "\u001b[0m\n",
      "\u001b[31m-\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mbuitinck, lars, gilles louppe, mathieu blondel, fabian pedregosa, andreas mueller,\n",
      "olivier grisel, vlad niculae, et al. 2013. “api design for machine learning software:\n",
      "experiences from the scikit-learn project.” arxiv preprint arxiv:1309.0238.\n",
      "\u001b[0m\n",
      "\u001b[31mivezić, željko, andrew j connolly, jacob t vanderplas, and alexander gray. 2014.\n",
      "statistics, data mining, and machine learning in astronomy: a practical python guide\n",
      "for the analysis of survey data. princeton university press.\n",
      "\u001b[0m\n",
      "\u001b[31mjones, eric, travis oliphant, pearu peterson, and others. 2001–2001--. “scipy: open\n",
      "source scientific tools for python.” http://www.scipy.org/.\n",
      "\u001b[0m\n",
      "\u001b[31mpedregosa, fabian, gaël varoquaux, alexandre gramfort, vincent michel, bertrand\n",
      "thirion, olivier grisel, mathieu blondel, et al. 2011. “scikit-learn: machine learn-\n",
      "ing in python.” the journal of machine learning research 12. jmlr. org: 2825–30.\n",
      "\u001b[0m\n",
      "\u001b[31mvanderplas, (2016). mst_clustering: clustering via euclidean minimum spanning trees. journal of open source software, 1(1), 12,\n",
      "doi:10.21105/joss.00012\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00016/10.21105.joss.00016.pdf\n",
      "\u001b[31mdoi: 10.21105/joss.00016\n",
      "\u001b[0m\n",
      "\u001b[31m1 university of southampton\n",
      "\u001b[0m\n",
      "\u001b[31malice harpole1 and ian hawke1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mr3d2: relativistic reactive riemann problem solver for\n",
      "deflagrations and detonations\n",
      "\u001b[0m\n",
      "\u001b[32mthis code extends standard exact solutions of the relativistic riemann problem to include\n",
      "a reaction term. it builds on existing solutions for the inert relativistic riemann problem,\n",
      "as described by (martí and müller 2015), and of the non-relativistic reactive riemann\n",
      "problem, as given by (zhang and zheng 1989).\n",
      "\u001b[0m\n",
      "\u001b[32mmodels of ideal hydrodynamics, where there is no viscosity or dissipation, can have solu-\n",
      "tions with discontinuities such as shocks. a simple case is the riemann problem, where\n",
      "two constant states are separated by a barrier. after the barrier is removed the solution\n",
      "develops, with waves (such as shocks and rarefactions) separating constant states.\n",
      "\u001b[0m\n",
      "\u001b[32mthe riemann problem has three main uses. efficient, often approximate, solvers are an\n",
      "integral part of many modern hydrodynamic evolution codes. second, the exact solution\n",
      "is a standard test for such codes. finally, the solver can illustrate features of discontinuous\n",
      "solutions in more complex scenarios.\n",
      "\u001b[0m\n",
      "\u001b[32min newtonian hydrodynamics, the riemann problem is one-dimensional: the solution\n",
      "depends only on the normal component of any vector quantities in the initial conditions.\n",
      "however, in relativistic systems, the lorentz factor introduces a coupling between the\n",
      "normal and tangential components. as found by (rezzolla and zanotti 2002), for high\n",
      "enough tangential velocities, the solution will smoothly transition from one wave pattern\n",
      "to another while maintaining the initial states otherwise unmodified. this code allows\n",
      "such transitions to be investigated in both inert and reactive systems.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mmartí, josé maria, and ewald müller. 2015. “grid-based methods in relativistic hydro-\n",
      "dynamics and magnetohydrodynamics.” living reviews in computational astrophysics 1.\n",
      "doi:10.1007/lrca-2015-3.\n",
      "\u001b[0m\n",
      "\u001b[31mrezzolla, luciano, and olindo zanotti.\n",
      "“new relativistic effects in the dy-\n",
      "namics of nonlinear hydrodynamical waves.” physical review letters 89 (11): 114501.\n",
      "doi:10.1103/physrevlett.89.114501.\n",
      "\u001b[0m\n",
      "\u001b[31m2002.\n",
      "\u001b[0m\n",
      "\u001b[31mzhang, tong, and yuxi zheng. 1989. “riemann problem for gasdynamic combustion.”\n",
      "journal of differential equations 77 (2): 203–30. doi:10.1016/0022-0396(89)90142-3.\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "harpole et al., (2016). r3d2: relativistic reactive riemann problem solver for deflagrations and detonations. journal of open source software,\n",
      "1(1), 16, doi:10.21105/joss.00016\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00017/10.21105.joss.00017.pdf\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00017\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mapplication skeleton: generating synthetic applications\n",
      "for infrastructure research\n",
      "\u001b[0m\n",
      "\u001b[31mzhao zhang1, daniel s. katz2, andre merzky3, matteo turilli3,\n",
      "shantenu jha3, and yadu nand4\n",
      "\u001b[0m\n",
      "\u001b[31m1 amplab and bids, university of california, berkeley 2 national center for supercomputing\n",
      "applications, university of illinois urbana-champaign 3 radical laboratory, rutgers university 4\n",
      "computation institute, university of chicago\n",
      "\u001b[0m\n",
      "\u001b[32mapplication skeleton is a simple and powerful tool to build simplified synthetic science\n",
      "and engineering applications (for example, modeling and simulation, data analysis) with\n",
      "runtime and i/o close to that of the real applications. it is intended for applied computer\n",
      "scientists who need to use science and engineering applications to verify the effectiveness\n",
      "of new systems designed to efficiently run such applications, so that they can bypass ob-\n",
      "stacles that they often encounter when accessing and building real science and engineering\n",
      "applications. using the applications generated by application skeleton guarantees that\n",
      "the cs systems’ effectiveness on synthetic applications will apply to the real applications.\n",
      "\u001b[0m\n",
      "\u001b[32mapplication skeleton can generate bag-of-task, (iterative) map-reduce, and (iterative) mul-\n",
      "tistage workflow applications. these applications are represented as a set of tasks, a set\n",
      "of input files, and a set of dependencies. these applications can be generally consid-\n",
      "ered many-task applications, and once created, can be run on single-core, single-node,\n",
      "multi-core, or multi-node (distributed or parallel) computers, depending on what work-\n",
      "flow system is used to run them. the generated applications are compatible with workflow\n",
      "system such as swift (zhao et al. 2007, wilde et al. (2009), wilde et al. (2011)) and\n",
      "pegasus (ewa deelman et al. 2004, e. deelman et al. (2005)), as well as the ubiquitous\n",
      "unix shell. the application can also be created as a generic json object that can be\n",
      "used by other systems such as the aimes (turilli et al. 2015) middleware.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mdeelman, e., g. singh, m.-h. su, j. blythe, y. gil, c. kesselman, g. mehta, et al. 2005.\n",
      "“pegasus: a framework for mapping complex scientific workflows onto distributed sys-\n",
      "tems.” scientific programming journal 13 (3): 219–37. doi:10.1155/2005/128026.\n",
      "\u001b[0m\n",
      "\u001b[31mdeelman, ewa, james blythe, yolanda gil, carl kesselman, gaurang mehta, sonal patil,\n",
      "mei-hui su, karan vahi, and miron livny. 2004. “pegasus: mapping scientific workflows\n",
      "onto the grid.” in grid computing, edited by marios d. dikaiakos, 3165:131–40. lect.\n",
      "notes in comp. sci. springer. doi:10.1007/978-3-540-28642-4_2.\n",
      "\u001b[0m\n",
      "\u001b[31mturilli, matteo, zhao zhang, andre merzky, michael wilde, jon weissman, daniel s katz,\n",
      "shantenu jha, and others. 2015. “integrating abstractions to enhance the execution of\n",
      "distributed applications.” arxiv preprint arxiv:1504.04720.\n",
      "\u001b[0m\n",
      "\u001b[31mwilde, michael, ian foster, kamil iskra, pete beckman, zhao zhang, allan espinosa,\n",
      "mihael hategan, ben clifford, and ioan raicu. 2009. “parallel scripting for applica-\n",
      "\u001b[0m\n",
      "\u001b[31mzhang et al., (2016). application skeleton: generating synthetic applications for infrastructure research. journal of open source software,\n",
      "1(1), 17, doi:10.21105/joss.00017\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n",
      "\u001b[32mtions at the petascale and beyond.” computer 42.\n",
      "doi:10.1109/mc.2009.365.\n",
      "\u001b[0m\n",
      "\u001b[32mieee computer society: 50–60.\n",
      "\u001b[0m\n",
      "\u001b[31mwilde, michael, mihael hategan, justin m. wozniak, ben clifford, daniel s. katz, and\n",
      "ian foster. 2011. “swift: a language for distributed parallel scripting.” par. comp.,\n",
      "september, 633–52. doi:10.1016/j.parco.2011.05.005.\n",
      "\u001b[0m\n",
      "\u001b[31mzhao, yong, mihael hategan, ben clifford, ian foster, gregor von laszewski, veronika\n",
      "nefedova, ioan raicu, tiberiu stef-praun, and michael wilde. 2007. “swift: fast, re-\n",
      "liable, loosely coupled parallel computation.” in ieee congress on services, 199–206.\n",
      "ieee. doi:10.1109/services.2007.63.\n",
      "\u001b[0m\n",
      "\u001b[31mzhang et al., (2016). application skeleton: generating synthetic applications for infrastructure research. journal of open source software,\n",
      "1(1), 17, doi:10.21105/joss.00017\n",
      "\u001b[0m\n",
      "\u001b[31m2\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00018/10.21105.joss.00018.pdf\n",
      "\u001b[32mxenomapper: mapping reads in a mixed species context\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00018\n",
      "\u001b[0m\n",
      "\u001b[31m1 the walter and eliza hall institute, the university of melbourne\n",
      "\u001b[0m\n",
      "\u001b[31mmatthew j. wakefield1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mxenomapper is a utility for post processing mapped dna sequencing reads that have\n",
      "been aligned to a primary genome and a secondary genome, and binning reads into species\n",
      "specific, multimapping in each species, unmapped and unassigned categories. it can be\n",
      "used on single end or paired end sequencing data across a wide range of genomics methods\n",
      "including rnaseq. in paired end data evidence of sequence specificity for either read will\n",
      "be used to assign both reads.\n",
      "\u001b[0m\n",
      "\u001b[32muse cases include xenografts of human cancers and host pathogen interactions.\n",
      "\u001b[0m\n",
      "\u001b[32mxenomapper is most effective with mapped reads that include an xs or zs score that\n",
      "gives the mapping score of the next best read. these include bowtie2 (langmead and\n",
      "salzberg 2012) and hisat (kim, langmead, and salzberg 2015).\n",
      "\u001b[0m\n",
      "\u001b[32mthis work builds upon a similar approach by rossello et. al. (rossello et al. 2013) with a\n",
      "more rigorous implementation and extensions for paired end data and exon aware aligners.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mkim, d., b. langmead, and s. l. salzberg. 2015. “hisat: a fast spliced aligner\n",
      "with low memory requirements.”\n",
      "journal article. nat methods 12 (4): 357–60.\n",
      "doi:10.1038/nmeth.3317.\n",
      "\u001b[0m\n",
      "\u001b[31mlangmead, b., and s. l. salzberg. 2012. “fast gapped-read alignment with bowtie 2.”\n",
      "journal article. nat methods 9 (4): 357–9. doi:10.1038/nmeth.1923.\n",
      "\u001b[0m\n",
      "\u001b[31mrossello, f. j., r. w. tothill, k. britt, k. d. marini, j. falzon, d. m. thomas, c. d.\n",
      "peacock, et al. 2013. “next-generation sequence analysis of cancer xenograft models.”\n",
      "journal article. plos one 8 (9): e74432. doi:10.1371/journal.pone.0074432.\n",
      "\u001b[0m\n",
      "\u001b[31mwakefield, (2016). xenomapper: mapping reads in a mixed species context. journal of open source software, 1(1), 18, doi:10.21105/joss.000181\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00020/10.21105.joss.00020.pdf\n",
      "\u001b[32mrepresentation and manipulation of genomic tuples in r\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00020\n",
      "\u001b[0m\n",
      "\u001b[31m1 department of biostatistics, johns hopkins bloomberg school of public health\n",
      "\u001b[0m\n",
      "\u001b[31mpeter f hickey1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mgenomictuples is an r/bioconductor package (r core team 2016; wolfgang huber\n",
      "et al 2015) that defines general purpose containers for storing and manipulating genomic\n",
      "tuples. a genomic tuple of size m is of the form chromosome:strand:{pos_1, pos_2,\n",
      "..., pos_m} where pos_1 < pos_2 < ... < pos_m are positions along the chromosome.\n",
      "the difference between a genomic tuple and a genomic range/interval is like that of a\n",
      "difference between an ordered set and an interval. for example, the genomic 2-tuple\n",
      "chr3:+:{65, 77} differs from the genomic range chr3:+:[65, 77] by not including any\n",
      "of the intervening loci, chr3:+:66 to chr3:+:76.\n",
      "\u001b[0m\n",
      "\u001b[32mgenomictuples aims to provide functionality for manipulating tuples of genomic co-\n",
      "ordinates that are analogous to those available for genomic ranges in the popular ge-\n",
      "nomicranges r/bioconductor package (lawrence et al. 2013). to that end, the ge-\n",
      "nomictuples api mimics that of genomicranges. by extending classes defined in the\n",
      "genomicranges package, objects from the genomictuples package may be used as\n",
      "drop-in replacements for objects from the genomicranges package. this ensures easy\n",
      "interoperability with other popular bioconductor packages, such as summarizedexper-\n",
      "iment (morgan et al. 2016), and the availability of common operations, such as finding\n",
      "overlaps between genomic tuples and genomic features of interest.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mlawrence, michael, wolfgang huber, hervé pagès, patrick aboyoun, marc carl-\n",
      "son, robert gentleman, martin morgan, and vincent carey.\n",
      "“software for\n",
      "computing and annotating genomic ranges.” plos computational biology 9 (8).\n",
      "doi:10.1371/journal.pcbi.1003118.\n",
      "\u001b[0m\n",
      "\u001b[31m2013.\n",
      "\u001b[0m\n",
      "\u001b[31mmorgan, martin, valerie obenchain, jim hester, and hervé pagès. 2016. summarized-\n",
      "experiment: summarizedexperiment container.\n",
      "\u001b[0m\n",
      "\u001b[31mr core team. 2016. r: a language and environment for statistical computing. vienna,\n",
      "austria: r foundation for statistical computing. https://www.r-project.org/.\n",
      "\u001b[0m\n",
      "\u001b[32mwolfgang huber et al. 2015. “orchestrating high-throughput genomic analysis with\n",
      "bioconductor.” nature methods 12 (2): 115–21. doi:10.1038/nmeth.3252.\n",
      "\u001b[0m\n",
      "\u001b[31mhickey, (2016). representation and manipulation of genomic tuples in r. journal of open source software, 1(1), 20, doi:10.21105/joss.000201\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00021/10.21105.joss.00021.pdf\n",
      "\u001b[32mpyuca: a python implementation of the unicode\n",
      "collation algorithm\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00021\n",
      "\u001b[0m\n",
      "\u001b[31m1 jktauber.com\n",
      "\u001b[0m\n",
      "\u001b[31mj. k. tauber1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[31mcollation, the sorting of strings, is an important part of computational work in corpus\n",
      "linguistics and digital humanities. lexicographical sorting, however, is rarely appropriate\n",
      "for languages other than english. the unicode consortium has developed the unicode\n",
      "collation algorithm (the unicode consortium (2015)) to solve this problem.\n",
      "\u001b[0m\n",
      "\u001b[32mpyuca is a python implementation of the unicode collation algorithm suitable for re-\n",
      "searchers doing text processing in python. it passes 100% of the uca conformance tests\n",
      "for unicode 5.2.0 (python 2.7) and 6.3.0 (python 3.3+) with a variable-weighting setting\n",
      "of non-ignorable.\n",
      "\u001b[0m\n",
      "\u001b[32mpyuca includes the default unicode collation element table (ducet) which provides a\n",
      "default collation suitable for many of the world’s scripts.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mthe unicode consortium. 2015. “the unicode consortium. unicode collation algorithm\n",
      "(unicode technical standard #10).” http://unicode.org/reports/tr10/.\n",
      "\u001b[0m\n",
      "\u001b[31mtauber, (2016). pyuca: a python implementation of the unicode collation algorithm. journal of open source software, 1(1), 21,\n",
      "doi:10.21105/joss.00021\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00023/10.21105.joss.00023.pdf\n",
      "\u001b[31mdoi: 10.21105/joss.00023\n",
      "\u001b[0m\n",
      "\u001b[31m1 rti international\n",
      "\u001b[0m\n",
      "\u001b[31mjames p balhoff1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mscowl: a scala dsl for programming with the owl api\n",
      "\u001b[0m\n",
      "\u001b[32mscowl is a domain-specific language (dsl) which provides a convenient scala-based syn-\n",
      "tax for composing owl (web ontology language, (w3c owl working group 2009))\n",
      "expressions and axioms. effective use of complex, community developed ontologies within\n",
      "research applications, such as those from the obo library (open biomedical ontologies,\n",
      "(smith et al. 2007)), often requires custom programmatic manipulation of ontologies and\n",
      "data, along with input to and output from automated reasoners. allowing deeply nested\n",
      "owl constructs to be coded in a readable, declarative style, scowl facilitates use cases at\n",
      "the interface of application programming and owl modeling: 1) tabular data ingest and\n",
      "transformation into owl; 2) ontology manipulation via programmatic generation of ax-\n",
      "ioms from existing content; and 3) reporting on ontology content using pattern matching\n",
      "on owl objects.\n",
      "\u001b[0m\n",
      "\u001b[32mwithin scala applications, scowl adds extension methods to native java owl api objects\n",
      "(“the owl api,” n.d.), supporting two syntax styles: the pseudo-english manchester\n",
      "syntax (horridge and patel-schneider 2009), as well as the owl functional syntax (motik\n",
      "et al. 2009). integration with the standard java owl api provides compatibility with\n",
      "multiple freely available owl reasoners.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mhorridge, matthew, and peter f patel-schneider. 2009. “owl 2 web ontology language\n",
      "manchester syntax.” https://www.w3.org/tr/owl2-manchester-syntax/.\n",
      "\u001b[0m\n",
      "\u001b[31mmotik, boris, peter f patel-schneider, bijan parsia, conrad bock, achille fokoue, peter\n",
      "haase, rinke hoekstra, ian horrocks, alan ruttenberg, and uli sattler. 2009. “owl 2\n",
      "web ontology language: structural specification and functional-style syntax.” https:\n",
      "//www.w3.org/tr/owl2-syntax/.\n",
      "\u001b[0m\n",
      "\u001b[31msmith, barry, michael ashburner, cornelius rosse, jonathan bard, william bug, werner\n",
      "ceusters, louis j goldberg, et al. 2007. “the obo foundry: coordinated evolution\n",
      "of ontologies to support biomedical data integration.” nature biotechnology 25 (11):\n",
      "1251–5. doi:10.1038/nbt1346.\n",
      "\u001b[0m\n",
      "\u001b[32m“the owl api.” n.d. http://owlapi.sourceforge.net/.\n",
      "\u001b[0m\n",
      "\u001b[32mw3c owl working group.\n",
      "overview.” https://www.w3.org/tr/owl2-overview/.\n",
      "\u001b[0m\n",
      "\u001b[31m2009.\n",
      "\u001b[0m\n",
      "\u001b[32m“owl 2 web ontology language document\n",
      "\u001b[0m\n",
      "\u001b[31mbalhoff, (2016). scowl: a scala dsl for programming with the owl api. journal of open source software, 1(1), 23, doi:10.21105/joss.000231\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00024/10.21105.joss.00024.pdf\n",
      "\u001b[32mcorner.py: scatterplot matrices in python\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00024\n",
      "\u001b[0m\n",
      "\u001b[31m1 sagan fellow, university of washington\n",
      "\u001b[0m\n",
      "\u001b[31mdaniel foreman-mackey1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mthis python module uses matplotlib (hunter 2007) to visualize multidimensional samples\n",
      "using a scatterplot matrix. in these visualizations, each one- and two-dimensional pro-\n",
      "jection of the sample is plotted to reveal covariances. corner was originally conceived to\n",
      "display the results of markov chain monte carlo simulations and the defaults are chosen\n",
      "with this application in mind but it can be used for displaying many qualitatively different\n",
      "samples.\n",
      "\u001b[0m\n",
      "\u001b[32mdevelopment of corner happens on github and any issues can be raised there (foreman-\n",
      "mackey 2016). corner has been used extensively in the astronomical literature and it\n",
      "has occasionally been cited as corner.py or using its previous name triangle.py. the\n",
      "source code for corner has been archived to zenodo and it has the doi (zenodo archive\n",
      "2016)\n",
      "\u001b[0m\n",
      "\u001b[32mthe following is a simple demonstration of a visualization made with corner:\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mforeman-mackey, daniel. 2016. “corner.py on github.” https://github.com/dfm/corner.\n",
      "py.\n",
      "\u001b[0m\n",
      "\u001b[31mhunter, john d. 2007. “matplotlib: a 2d graphics environment.” computing in science\n",
      "and engineering 9 (3): 90–95. doi:10.1109/mcse.2007.55.\n",
      "\u001b[0m\n",
      "\u001b[31mzenodo archive. 2016. “corner.py: scatterplot matrices in python.” http://dx.doi.org/\n",
      "10.5281/zenodo.53155. doi:10.5281/zenodo.53155.\n",
      "\u001b[0m\n",
      "\u001b[31mforeman-mackey, (2016). corner.py: scatterplot matrices in python. journal of open source software, 1(2), 24, doi:10.21105/joss.00024\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n",
      "\u001b[31mfigure 1: a scatterplot matrix generated by corner.\n",
      "\u001b[0m\n",
      "\u001b[31mforeman-mackey, (2016). corner.py: scatterplot matrices in python. journal of open source software, 1(2), 24, doi:10.21105/joss.00024\n",
      "\u001b[0m\n",
      "\u001b[31m2\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00025/10.21105.joss.00025.pdf\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00025\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mgenenetwork: framework for web-based genetics\n",
      "\u001b[0m\n",
      "\u001b[31mzachary sloan1, danny arends2, karl w. broman4, arthur centeno1,\n",
      "nicholas furlotte1, harm nijveen5, lei yan1, xiang zhou6, robert w.\n",
      "williams1, and pjotr prins3,1\n",
      "\u001b[0m\n",
      "\u001b[31m1 university of tennessee health science center, usa 2 humboldt university, berlin, germany 3\n",
      "university medical center utrecht, the netherlands 4 university of wisconsin, usa 5 wageningen\n",
      "university, the netherlands 6 university of michigan\n",
      "\u001b[0m\n",
      "\u001b[32mgenenetwork (gn) is a free and open source (foss) framework for web-based genetics\n",
      "that can be deployed anywhere. gn allows biologists to upload high-throughput exper-\n",
      "imental data, such as expression data from microarrays and rna-seq, and also ‘classic’\n",
      "phenotypes, such as disease phenotypes. these phenotypes can be mapped interactively\n",
      "against genotypes using embedded tools, such as r/qtl (arends et al. 2010) mapping,\n",
      "interval mapping for model organisms and pylmm; an implementation of fast-lmm (lip-\n",
      "pert et al. 2011) which is more suitable for human populations and outbred crosses, such\n",
      "as the mouse diversity outcross. interactive d3 graphics are included from r/qtlcharts\n",
      "and presentation-ready figures can be generated. recently we have added functionality for\n",
      "phenotype correlation (wang et al. 2016) and network analysis (langfelder and horvath\n",
      "2008).\n",
      "\u001b[0m\n",
      "\u001b[32mgn is written in python and javascript and contains a rich set of tools and libraries that\n",
      "can be written in any computer language. a full list of included software can be found in\n",
      "the package named ‘genenetwork2’ and defined in guix-bioinformatics. to make it easy\n",
      "to install gn locally in a byte reproducible way, including all dependencies and a 2gb\n",
      "mysql test database (the full database is 160gb and growing), gn is packaged with\n",
      "gnu guix, as described here. gnu guix deployment makes it feasible to deploy and\n",
      "rebrand gn anywhere.\n",
      "\u001b[0m\n",
      "\u001b[31mfuture work\n",
      "\u001b[0m\n",
      "\u001b[32mmore mapping tools will be added, including support for genome-wide efficient mixed\n",
      "model association (gemma). the biodiallance genome browser is being added as a\n",
      "google summer of code project with special tracks related to qtl mapping and network\n",
      "analysis. faster lmm solutions are being worked on, including gpu support.\n",
      "\u001b[0m\n",
      "\u001b[32ma rest interface is being added so that data can be uploaded to a server, analysis run\n",
      "remotely on high performance hardware, and results downloaded and used for further\n",
      "analysis. this feature will allow biologist-programmers to use r and python on their\n",
      "computer and execute computations on gn enabled servers.\n",
      "\u001b[0m\n",
      "\u001b[31msloan et al., (2016). genenetwork: framework for web-based genetics. journal of open source software, 1(2), 25, doi:10.21105/joss.00025\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n",
      "\u001b[31mfigure 1: mouse lmm mapping example\n",
      "\u001b[0m\n",
      "\u001b[31msloan et al., (2016). genenetwork: framework for web-based genetics. journal of open source software, 1(2), 25, doi:10.21105/joss.00025\n",
      "\u001b[0m\n",
      "\u001b[31m2\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31marends, d., p. prins, r. c. jansen, and k. w. broman. 2010. “r/qtl: high-throughput\n",
      "multiple qtl mapping.” bioinformatics 26 (23): 2990–2. doi:10.1093/bioinformatics/btq565.\n",
      "\u001b[0m\n",
      "\u001b[31mlangfelder, p., and s. horvath. 2008. “wgcna: an r package for weighted correlation\n",
      "network analysis.” bmc bioinformatics 9: 559. doi:10.1186/1471-2105-9-559.\n",
      "\u001b[0m\n",
      "\u001b[31mlippert, c., j. listgarten, y. liu, c. m. kadie, r. i. davidson, and d. heckerman. 2011.\n",
      "“fast linear mixed models for genome-wide association studies.” nat methods 8 (10):\n",
      "833–35. doi:10.1038/nmeth.1681.\n",
      "\u001b[0m\n",
      "\u001b[31mwang, x., a. k. pandey, m. k. mulligan, e. g. williams, k. mozhui, z. li, v. jovaisaite,\n",
      "et al. 2016. “joint mouse-human phenome-wide association to test gene function and\n",
      "disease risk.” nat commun 7: 10464. doi:10.1038/ncomms10464.\n",
      "\u001b[0m\n",
      "\u001b[31msloan et al., (2016). genenetwork: framework for web-based genetics. journal of open source software, 1(2), 25, doi:10.21105/joss.00025\n",
      "\u001b[0m\n",
      "\u001b[31m3\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00026/10.21105.joss.00026.pdf\n",
      "\u001b[32marmadillo: a template-based c++ library for linear\n",
      "algebra\n",
      "\u001b[0m\n",
      "\u001b[31mconrad sanderson1 and ryan curtin2\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00026\n",
      "\u001b[0m\n",
      "\u001b[31m1 csiro, australia, and nicta, australia 2 symantec corporation, usa\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mthe c++ language is often used for implementing functionality that is performance\n",
      "and/or resource sensitive. while the standard c++ library provides many useful al-\n",
      "gorithms (such as sorting), in its current form it does not provide direct handling of\n",
      "linear algebra (matrix maths).\n",
      "\u001b[0m\n",
      "\u001b[32marmadillo is an open source linear algebra library for the c++ language, aiming towards\n",
      "a good balance between speed and ease of use. its high-level application programming\n",
      "interface (api) is deliberately similar to the widely matlab and octave languages (eaton\n",
      "et al. 2015), so that mathematical operations can be expressed in a familiar and natural\n",
      "manner. the library is useful for algorithm development directly in c++, or relatively\n",
      "quick conversion of research code into production environments.\n",
      "\u001b[0m\n",
      "\u001b[32marmadillo provides efficient objects for vectors, matrices and cubes (third order tensors),\n",
      "as well as over 200 associated functions for manipulating data stored in the objects. in-\n",
      "teger, floating point and complex numbers are supported, as well as dense and sparse\n",
      "storage formats. various matrix factorisations are provided through integration with la-\n",
      "pack (demmel 1997), or one of its high performance drop-in replacements such as intel\n",
      "mkl (intel 2016) or openblas (xianyi, qian, and saar 2016). it is also possible to use\n",
      "armadillo in conjunction with nvblas to obtain gpu-accelerated matrix multiplication\n",
      "(nvidia 2015).\n",
      "\u001b[0m\n",
      "\u001b[32marmadillo is used as a base for other open source projects, such as mlpack, a c++\n",
      "library for machine learning and pattern recognition (curtin et al. 2013), and rcppar-\n",
      "madillo, a bridge between the r language and c++ in order to speed up computations\n",
      "(eddelbuettel and sanderson 2014).\n",
      "\u001b[0m\n",
      "\u001b[32marmadillo internally employs an expression evaluator based on template meta-\n",
      "programming techniques (abrahams and gurtovoy 2004), to automatically combine\n",
      "several operations in order to increase speed and efficiency. an overview of the internal\n",
      "architecture is given in (sanderson 2010).\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mabrahams, david, and aleksey gurtovoy. 2004. c++ template metaprogramming: con-\n",
      "cepts, tools, and techniques from boost and beyond. addison-wesley professional.\n",
      "\u001b[0m\n",
      "\u001b[31mcurtin, ryan r., james r. cline, n. p. slagle, william b. march, parikshit ram, nishant\n",
      "a. mehta, and alexander g. gray. 2013. “mlpack: a scalable c++ machine learning\n",
      "library.” journal of machine learning research 14 (mar): 801–5. http://jmlr.org/\n",
      "\u001b[0m\n",
      "\u001b[31msanderson et al., (2016). armadillo: a template-based c++ library for linear algebra. journal of open source software, 1(2), 26,\n",
      "doi:10.21105/joss.00026\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n",
      "\u001b[31mpapers/v14/curtin13a.html.\n",
      "\u001b[0m\n",
      "\u001b[31mdemmel, james w. 1997. applied numerical linear algebra. siam.\n",
      "\u001b[0m\n",
      "\u001b[31meaton, john w., david bateman, søren hauberg, and rik wehbring. 2015. gnu octave\n",
      "version 4.0.0 manual: a high-level interactive language for numerical computations.\n",
      "http://www.gnu.org/software/octave/.\n",
      "\u001b[0m\n",
      "\u001b[31meddelbuettel, dirk, and conrad sanderson. 2014. “rcpparmadillo: accelerating r with\n",
      "high-performance c++ linear algebra.” computational statistics & data analysis 71:\n",
      "1054–63. doi:10.1016/j.csda.2013.02.005.\n",
      "\u001b[0m\n",
      "\u001b[31mintel. 2016. “math kernel library (mkl).” http://software.intel.com/en-us/intel-mkl/.\n",
      "\u001b[0m\n",
      "\u001b[31mnvidia. 2015. “nvblas library.” http://docs.nvidia.com/cuda/nvblas/.\n",
      "\u001b[0m\n",
      "\u001b[31msanderson, conrad. 2010. “armadillo: an open source c++ linear algebra library for\n",
      "fast prototyping and computationally intensive experiments.” nicta.\n",
      "\u001b[0m\n",
      "\u001b[31mxianyi, zhang, wang qian, and werner saar. 2016. “openblas: an optimized blas\n",
      "library.” http://www.openblas.net/.\n",
      "\u001b[0m\n",
      "\u001b[31msanderson et al., (2016). armadillo: a template-based c++ library for linear algebra. journal of open source software, 1(2), 26,\n",
      "doi:10.21105/joss.00026\n",
      "\u001b[0m\n",
      "\u001b[31m2\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00027/10.21105.joss.00027.pdf\n",
      "\u001b[32msourmash: a library for minhash sketching of dna\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00027\n",
      "\u001b[0m\n",
      "\u001b[31m1 university of california, davis\n",
      "\u001b[0m\n",
      "\u001b[31mc. titus brown1 and luiz irber1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32msourmash is a toolbox for creating, comparing, and manipulating minhash sketches of\n",
      "genomic data.\n",
      "\u001b[0m\n",
      "\u001b[32mminhash sketches provide a lightweight way to store “signatures” of large dna or rna\n",
      "sequence collections, and then compare or search them using a jaccard index. minhash\n",
      "sketches can be used to identify samples, find similar samples, identify data sets with\n",
      "shared sequences, and build phylogenetic trees (ondov et al. 2015).\n",
      "\u001b[0m\n",
      "\u001b[32msourmash provides a command line script, a python library, and a cpython module for\n",
      "minhash sketches.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mondov, brian d, todd j treangen, adam b mallonee, nicholas h bergman, sergey koren,\n",
      "and adam m phillippy. 2015. “fast genome and metagenome distance estimation using\n",
      "minhash.” biorxiv. cold spring harbor labs journals, 029827. doi:10.1101/029827.\n",
      "\u001b[0m\n",
      "\u001b[31mbrown et al., (2016). sourmash: a library for minhash sketching of dna. journal of open source software, 1(5), 27, doi:10.21105/joss.00027 1\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00028/10.21105.joss.00028.pdf\n",
      "\u001b[32mphylogemetric: a python library for calculating\n",
      "phylogenetic network metrics\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00028\n",
      "\u001b[0m\n",
      "\u001b[31m1 arc centre of excellence for the dynamics of language, australian national university\n",
      "\u001b[0m\n",
      "\u001b[31msimon j. greenhill1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mphylogemetric is a python library for calculating the (cid:14)-score (holland et al. 2002) and\n",
      "q-residual (gray, bryant, and greenhill 2010) for phylogenetic data. these methods are\n",
      "used in studies of linguistic and cultural evolution to quantify reticulation in data.\n",
      "\u001b[0m\n",
      "\u001b[32mthis python library provides a command-line script interface for use on nexus-formatted\n",
      "data (d. r. maddison, swofford, and maddison 1997), and an importable python library\n",
      "for use on any binary matrix.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mgray, russell d., david bryant, and simon j. greenhill. 2010. “on the shape and fabric\n",
      "of human history.” philosophical transactions of the royal society of london. series b,\n",
      "biological sciences 365 (1559): 3923–33. doi:10.1098/rstb.2010.0162.\n",
      "\u001b[0m\n",
      "\u001b[31mholland, barbara r, k t huber, a dress, and v moulton. 2002. “[delta] plots: a tool for\n",
      "analyzing phylogenetic distance data.” molecular biology and evolution 19 (12): 2051–9.\n",
      "http://mbe.oxfordjournals.org/content/19/12/2051.full.\n",
      "\u001b[0m\n",
      "\u001b[31mmaddison, d r, d l swofford, and wayne p. maddison. 1997. “nexus: an extensible\n",
      "file format for systematic information.” systematic biology 46 (4): 590–621.\n",
      "\u001b[0m\n",
      "\u001b[31mgreenhill, (2016). phylogemetric: a python library for calculating phylogenetic network metrics. journal of open source software, 1(2), 28,\n",
      "doi:10.21105/joss.00028\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00029/10.21105.joss.00029.pdf\n",
      "\u001b[32mgit-rdm: a research data management plugin for the\n",
      "git version control system\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00029\n",
      "\u001b[0m\n",
      "\u001b[31m1 university of southampton 2 imperial college london\n",
      "\u001b[0m\n",
      "\u001b[31mchristian t. jacobs1 and alexandros avdis2\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mmany research funding agencies (research council uk 2015) and research societies (royal\n",
      "society 2012) are increasingly requiring that data from at least publicly funded research be\n",
      "made openly available, and with clear citations that describe provenance. these require-\n",
      "ments have led to the proliferation of institutional repositories with universities maintain-\n",
      "ing a handful of data services, but also repository services capable of minting a persistent\n",
      "and citable digital object identifier (doi) (technical committee iso/tc 46 (informa-\n",
      "tion and documentation), subcommittee sc 9 (identification and description) 2012) for\n",
      "every published item. figshare (figshare.com) and zenodo (zenodo.org) are examples of\n",
      "the latter. alongside data, software is also increasingly seen as a research output. this\n",
      "viewpoint necessitates not just open-source publication of code, but also provenance and\n",
      "attribution. while a doi is an identifier of static items, many research teams use version\n",
      "control systems and services to organise their collective efforts and publish output, be\n",
      "that code or data. popular examples include git (chacon and straub 2014) and github\n",
      "(github.com).\n",
      "\u001b[0m\n",
      "\u001b[32mgit-rdm is a research data management (rdm) plugin for the git version control\n",
      "system. it interfaces git with data hosting services to manage the curation of version\n",
      "controlled files using persistent, citable repositories. this facilitates the sharing of research\n",
      "outputs and encourages a more open workflow within the research community.\n",
      "\u001b[0m\n",
      "\u001b[32mmuch like the standard git commands, git-rdm allows users to add/remove files within\n",
      "a ‘publication staging area’. when ready, users can readily publish these staged files to a\n",
      "data repository hosted either by figshare or zenodo via the command line; this curation\n",
      "step is handled by the pyrdm library (jacobs et al. 2014). details of the files and\n",
      "their associated publication(s) are then recorded in a local sqlite database, including\n",
      "the specific git revision (in the form of a sha-1 hash), publication date/time, and the\n",
      "doi, such that a full history of data publication is maintained.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mchacon, s., and b. straub. 2014. pro git. 2nd ed. apress.\n",
      "\u001b[0m\n",
      "\u001b[31mjacobs, c. t., a. avdis, g. j. gorman, and m. d. piggott. 2014. “pyrdm: a python-\n",
      "based library for automating the management and online publication of scientific software\n",
      "and data.” journal of open research software 2 (1): e28. doi:10.5334/jors.bj.\n",
      "\u001b[0m\n",
      "\u001b[31mresearch council uk. 2015. “guidance on best practice in the management of research\n",
      "data.”\n",
      "\u001b[0m\n",
      "\u001b[31mroyal society. 2012. “science as an open enterprise: open data for open science.”\n",
      "\u001b[0m\n",
      "\u001b[32mtechnical committee iso/tc 46 (information and documentation), subcommittee sc 9\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "jacobs et al., (2016). git-rdm: a research data management plugin for the git version control system. journal of open source software, 1(2),\n",
      "29, doi:10.21105/joss.00029\n",
      "\u001b[0m\n",
      "\u001b[31m(identification and description). 2012. “iso 26324:2012 information and documentation\n",
      "– digital object identifier system.” international organisation for standardization.\n",
      "\u001b[0m\n",
      "\u001b[31m2\n",
      "jacobs et al., (2016). git-rdm: a research data management plugin for the git version control system. journal of open source software, 1(2),\n",
      "29, doi:10.21105/joss.00029\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00031/10.21105.joss.00031.pdf\n",
      "\u001b[32mprism: multiple spline regression with regularization,\n",
      "dimensionality reduction, and feature selection\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00031\n",
      "\u001b[0m\n",
      "\u001b[31m1 boston college\n",
      "\u001b[0m\n",
      "\u001b[31mchristopher r madan1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mprism uses a combination of statistical methods to conduct spline-based multiple re-\n",
      "gression. prism conducts this regression using regularization, dimensionality reduction,\n",
      "and feature selection, through a combination of smoothing spline regression, pca, and\n",
      "rvr/lasso. smoothing splines can be used to model non-parametric relationships using\n",
      "piece-wise cubic functions (wahba and wold 1975; fox 2000). relevance vector regression\n",
      "(rvr) refers to application of a relevance vector machine (rvm) to a regression problem;\n",
      "broadly, rvm is similar to multiple linear regression with regularization, using automatic\n",
      "relevance determination for feature selection (tipping 2000). rvm shares many common-\n",
      "alities with svm, and is implemented as a special case of a sparse bayesian framework\n",
      "(tipping 2001; tipping and faul 2003).\n",
      "\u001b[0m\n",
      "\u001b[32mprism has been tested in matlab 2015b and requires three first-party toolboxes: (1)\n",
      "curve fitting toolbox; (2) statistics and machine learning toolbox; (3) signal process-\n",
      "ing toolbox. relevance vector regression requires the sparsebayes v2 toolbox, which can\n",
      "be obtained from http://www.relevancevector.com.\n",
      "\u001b[0m\n",
      "\u001b[31mfigure 1: illustration of prism regression procedure, first conducting spline regression\n",
      "for each predictor, followed by dimensionality reduction and feature selection (panel a).\n",
      "the logo for prism is shown in panel b.\n",
      "\u001b[0m\n",
      "\u001b[32mprism was implemented for conducting multiple regression investigating age-related dif-\n",
      "ferences in brain morphology. while linear and quadratic regression are often used (e.g.,\n",
      "sowell et al. (2003); hogstrom et al. (2013); madan and kensinger (2016)), it has been\n",
      "shown that non-linear (spline) regression is more appropriate (fjell et al. 2010; fjell et\n",
      "al. 2013).\n",
      "\u001b[0m\n",
      "\u001b[31mmadan, (2016). prism: multiple spline regression with regularization, dimensionality reduction, and feature selection. journal of open source\n",
      "software, 1(3), 31, doi:10.21105/joss.00031\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n",
      "\u001b[32mexample implementation\n",
      "\u001b[0m\n",
      "\u001b[32mto provide an example of prism’s implementation, see the ‘demo’ folder. demo.mat was\n",
      "generated using brain morphology data, where each column corresponds to a brain mor-\n",
      "phology measure. the test and training data are derived from independent mr sites,\n",
      "and the dependent variable is age. the data is from an unpublished study and the pre-\n",
      "processing pipeline is not finalized (e.g., correcting for site effects), so further information\n",
      "of the dataset is not provided at this time.\n",
      "\u001b[0m\n",
      "\u001b[31mfigure 2: output scatter plots for code in demo.m.\n",
      "\u001b[0m\n",
      "\u001b[32msee readme.md for further details.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mfjell, a m, k b walhovd, l t westlye, y østby, c k tamnes, t l jernigan,\n",
      "“when does brain aging accelerate? dan-\n",
      "a gamst, and a m dale.\n",
      "1376–83.\n",
      "gers of quadratic fits in cross-sectional studies.”\n",
      "doi:10.1016/j.neuroimage.2010.01.061.\n",
      "\u001b[0m\n",
      "\u001b[32mneuroimage 50:\n",
      "\u001b[0m\n",
      "\u001b[31m2010.\n",
      "\u001b[0m\n",
      "\u001b[31mfjell, a m, l t westlye, h grydeland, i amlien, t espeseth, i reinvang, n raz, d\n",
      "holland, a m.dale, and k b walhovd. 2013. “critical ages in the life course of\n",
      "the adult brain: nonlinear subcortical aging.” neurobiology of aging 34: 2239–47.\n",
      "doi:10.1016/j.neurobiolaging.2013.04.006.\n",
      "\u001b[0m\n",
      "\u001b[31mfox, j. 2000. nonparametric simple regression: smoothing scatterplots. sage.\n",
      "\u001b[0m\n",
      "\u001b[31mhogstrom, l j, l t westlye, k b walhovd, and a m fjell. 2013. “the structure of\n",
      "the cerebral cortex across adult life: age-related patterns of surface area, thickness,\n",
      "and gyrification.” cerebral cortex 23: 2521–30. doi:10.1093/cercor/bhs231.\n",
      "\u001b[0m\n",
      "\u001b[31mmadan, c r, and e a kensinger. 2016. “cortical complexity as a measure of age-related\n",
      "brain atrophy.” neuroimage 134: 617–29. doi:10.1016/j.neuroimage.2016.04.029.\n",
      "\u001b[0m\n",
      "\u001b[31msowell, e r, b s peterson, p m thompson, s e welcome, a l henkenius, and a w toga.\n",
      "2003. “mapping cortical change across the human life span.” nature neuroscience 6:\n",
      "309–15. doi:10.1038/nn1008.\n",
      "\u001b[0m\n",
      "\u001b[31mtipping, m e. 2000. “the relevance vector machine.” advances in neural information\n",
      "processing systems (nips), 652–58.\n",
      "\u001b[0m\n",
      "\u001b[31m———. 2001. “sparse bayesian learning and the relevance vector machine.” journal of\n",
      "machine learning research 1: 211–44.\n",
      "\u001b[0m\n",
      "\u001b[31mtipping, m e, and a c faul. 2003. “fast marginal likelihood maximisation for sparse\n",
      "\u001b[0m\n",
      "\u001b[31mmadan, (2016). prism: multiple spline regression with regularization, dimensionality reduction, and feature selection. journal of open source\n",
      "software, 1(3), 31, doi:10.21105/joss.00031\n",
      "\u001b[0m\n",
      "\u001b[31m2\n",
      "\u001b[0m\n",
      "\u001b[32mbayesian models.” proceedings of the ninth international workshop on artificial intelli-\n",
      "gence and statistics (aistats).\n",
      "\u001b[0m\n",
      "\u001b[32m“a completely automatic french curve: fit-\n",
      "wahba, g, and s wold.\n",
      "ting spline functions by cross validation.” communications in statistics 4: 1–17.\n",
      "doi:10.1080/03610927508827223.\n",
      "\u001b[0m\n",
      "\u001b[31m1975.\n",
      "\u001b[0m\n",
      "\u001b[31mmadan, (2016). prism: multiple spline regression with regularization, dimensionality reduction, and feature selection. journal of open source\n",
      "software, 1(3), 31, doi:10.21105/joss.00031\n",
      "\u001b[0m\n",
      "\u001b[31m3\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00032/10.21105.joss.00032.pdf\n",
      "\u001b[32madvanced viewshed analysis: a quantum gis plug-in for\n",
      "the analysis of visual landscapes\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00032\n",
      "\u001b[0m\n",
      "\u001b[31m1 umr 6249 laboratoire chrono-environnement, université bourgogne franche-comté\n",
      "\u001b[0m\n",
      "\u001b[31mzoran cuckovic1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mviewshed analysis is a standard feature of gis software packages, such as arcgis (esri\n",
      "2016), grass (neteler et al. 2012) or erdas (2015). however, these implementations\n",
      "vary considerably in terms of their versatility and robustness. software in the free domain\n",
      "is particularly poor in this respect: visibility analysis is generally implemented as a simple\n",
      "binary query (true/false) for elevation datasets (eg. grass or saga gis). in order\n",
      "to meet the demands of a typical analysis concerning visual landscapes we would be\n",
      "interested to find out how deep are certain locations below the visible horizon, what is\n",
      "the overall visual potential of a landscape or which sites are connected in visual networks\n",
      "(cf. higuchi (1983); llobera (2003); čučković (2015)).\n",
      "\u001b[0m\n",
      "\u001b[32madvanced viewshed analysis plug-in for open source quantum gis software has been made\n",
      "in order to meet some of these demands: besides standard binary viewshed it provides\n",
      "information on the depth at which objects may be hidden from view, mapping of visual\n",
      "horizon and analysis of intervisibility networks. as of version 0.5.1, the implemented\n",
      "algorithm has been adapted for intensive, repetitive viewshed calculation from multiple\n",
      "observation points.\n",
      "\u001b[0m\n",
      "\u001b[32mthe plug-in is coded in python and is dependant of the quantum gis framework. more\n",
      "specifically, it makes use of following libraries (bundled with quantum gis): numpy, gdal\n",
      "and qgis core library.\n",
      "\u001b[0m\n",
      "\u001b[32man example of intervisibility output:\n",
      "\u001b[0m\n",
      "\u001b[31mcuckovic, (2016). advanced viewshed analysis: a quantum gis plug-in for the analysis of visual landscapes. journal of open source software,\n",
      "1(4), 32, doi:10.21105/joss.00032\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n",
      "\u001b[31mfigure 1: intervisibility network\n",
      "\u001b[0m\n",
      "\u001b[31mcuckovic, (2016). advanced viewshed analysis: a quantum gis plug-in for the analysis of visual landscapes. journal of open source software,\n",
      "1(4), 32, doi:10.21105/joss.00032\n",
      "\u001b[0m\n",
      "\u001b[31m2\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mčučković, zoran. 2015. “exploring intervisibility networks: a case study from bronze\n",
      "and iron age istria (croatia and slovenia).” in caa 2014 - 21st century archaeology :\n",
      "proceedings of the 42nd annual conference on computer applications and quantitative\n",
      "methods in archaeology, edited by françois giligny, françois djindjian, laurent costa,\n",
      "paola moscati, and sandrine robert, 469–478. paris, france: archaeopress.\n",
      "\u001b[0m\n",
      "\u001b[31merdas. 2015. “erdas imagine suite 2015.” norcross: hexagon geospatial.\n",
      "\u001b[0m\n",
      "\u001b[31mesri. 2016. “arcgis desktop: release 10.4.” redlands: environmental systems re-\n",
      "search institute.\n",
      "\u001b[0m\n",
      "\u001b[31mhiguchi, tadahiko. 1983. visual and spatial structure of landscapes. cambridge, mass.:\n",
      "mit press.\n",
      "\u001b[0m\n",
      "\u001b[31mllobera, marcos. 2003. “extending gis-based visual analysis: the concept of vi-\n",
      "sualscapes.” international journal of geographical information science 17 (1): 25–48.\n",
      "doi:10.1080/713811741.\n",
      "\u001b[0m\n",
      "\u001b[31mneteler, markus, m. hamish bowman, martin landa, and markus metz. 2012. “grass\n",
      "gis: a multi-purpose open source gis.” environmental modelling & software 31: 124–30.\n",
      "doi:http://dx.doi.org/10.1016/j.envsoft.2011.11.014.\n",
      "\u001b[0m\n",
      "\u001b[31mcuckovic, (2016). advanced viewshed analysis: a quantum gis plug-in for the analysis of visual landscapes. journal of open source software,\n",
      "1(4), 32, doi:10.21105/joss.00032\n",
      "\u001b[0m\n",
      "\u001b[31m3\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00033/10.21105.joss.00033.pdf\n",
      "\u001b[32mrsmtool: collection of tools building and evaluating\n",
      "automated scoring models\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00033\n",
      "\u001b[0m\n",
      "\u001b[31m1 educational testing service\n",
      "\u001b[0m\n",
      "\u001b[31mnitin madnani1 and anastassia loukina1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mrsmtool is a collection of tools for researchers working on the development of auto-\n",
      "mated scoring engines for written and spoken responses in an educational context. the\n",
      "main purpose of the tool is to simplify the integration of educational measurement recom-\n",
      "mendations into the model building process and to allow the researchers to rapidly and\n",
      "automatically generate comprehensive evaluations of the scoring model.\n",
      "\u001b[0m\n",
      "\u001b[32mrsmtool takes as input a feature file with numeric, non-sparse features extracted from the\n",
      "responses and a human score and lets you try several different machine learning algorithms\n",
      "to try and predict the human score from the features. rsmtool allows the use of simple\n",
      "ols regression as well as several more sophisticated regressors including ridge, svr,\n",
      "adaboost, and random forests, available through integration with the skll toolkit\n",
      "(blanchard et al. 2014). the tool also includes several regressors which ensure that\n",
      "all coefficients in the final model are positive to meet the requirement that all feature\n",
      "contributions are additive (loukina et al. 2015). the primary novel contribution of\n",
      "rsmtool is a comprehensive, customizable html statistical report that contains feature\n",
      "descriptives, subgroup analyses, model statistics, as well as several different evaluation\n",
      "measures illustrating model efficacy. the various numbers and figures in the report are\n",
      "highlighted based on whether they exceed or fall short of the recommendations laid out\n",
      "by (williamson, xi, and breyer 2012). the structure of rsmtool makes it easy for\n",
      "researchers to add new analyses without making any changes to the core code structure\n",
      "thus allowing for a wide range of psychometric evaluations.\n",
      "\u001b[0m\n",
      "\u001b[32mthe tool is written in python and works on all platforms (windows/linux/mac os x).\n",
      "the source code is available from github (madnani and loukina 2016)\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mblanchard, dan, nitin madnani, michael heilman, nils murrugarra llerena, diane m.\n",
      "napolitano, aoife cahill, keelan evanini, and chee wee leong. 2014. “scikit-learn\n",
      "laboratory (skll) 1.0.0.” doi:10.5281/zenodo.12825.\n",
      "\u001b[0m\n",
      "\u001b[31mloukina, anastassia, klaus zechner, lei chen, and michael heilman. 2015. “feature\n",
      "selection for automated speech scoring.” in proceedings of the tenth workshop on inno-\n",
      "vative use of nlp for building educational applications, 12–19. denver, colorado.\n",
      "\u001b[0m\n",
      "\u001b[31mmadnani, nitin, and anastassia loukina.\n",
      "educationaltestingservice/rsmtool.\n",
      "\u001b[0m\n",
      "\u001b[31m2016.\n",
      "\u001b[0m\n",
      "\u001b[32m“rsmtool.” https://github.com/\n",
      "\u001b[0m\n",
      "\u001b[31mwilliamson, david m., xiaoming xi, and f. jay breyer. 2012. “a framework for evalu-\n",
      "ation and use of automated scoring.” educational measurement: issues and practice 31\n",
      "(1): 2–13. doi:10.1111/j.1745-3992.2011.00223.x.\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "madnani et al., (2016). rsmtool: collection of tools building and evaluating automated scoring models. journal of open source software, 1(3),\n",
      "33, doi:10.21105/joss.00033\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00034/10.21105.joss.00034.pdf\n",
      "\u001b[31mdoi: 10.21105/joss.00034\n",
      "\u001b[0m\n",
      "\u001b[31m1 stanford university 2 jagiellonian university\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mosprey: hyperparameter optimization for machine\n",
      "learning\n",
      "\u001b[0m\n",
      "\u001b[32mrobert t. mcgibbon1, carlos x. hernández1, matthew p. harrigan1,\n",
      "steven kearnes1, mohammad m. sultan1, stanislaw jastrzebski2,\n",
      "brooke e. husic1, and vijay s. pande1\n",
      "\u001b[0m\n",
      "\u001b[32mosprey is a tool for hyperparameter optimization of machine learning algorithms in python.\n",
      "hyperparameter optimization can often be an onerous process for researchers, due to time-\n",
      "consuming experimental replicates, non-convex objective functions, and constant tension\n",
      "between exploration of global parameter space and local optimization (jones, schonlau,\n",
      "and welch 1998). we’ve designed osprey to provide scientists with a practical, easy-\n",
      "to-use way of finding optimal model parameters. the software works seamlessly with\n",
      "scikit-learn estimators (pedregosa et al. 2011) and supports many different search\n",
      "strategies for choosing the next set of parameters with which to evaluate a given model,\n",
      "including gaussian processes (gpy 2012), tree-structured parzen estimators (yamins, tax,\n",
      "and bergstra 2013), as well as random and grid search. as hyperparameter optimization\n",
      "is an embarrassingly parallel problem, osprey can easily scale to hundreds of concurrent\n",
      "processes by executing a simple command-line program multiple times. this makes it\n",
      "easy to exploit large resources available in high-performance computing environments.\n",
      "\u001b[0m\n",
      "\u001b[32mosprey is actively maintained by researchers at stanford university and other institutions\n",
      "around the world. while originally developed to analyze computational protein dynamics\n",
      "(mcgibbon, harrigan, et al. 2016), it is applicable to any scikit-learn-compatible\n",
      "pipeline. the source code for osprey is hosted on github and has been archived to\n",
      "zenodo (mcgibbon, hernández, et al. 2016). full documentation can be found at http:\n",
      "//msmbuilder.org/osprey.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mgpy. 2012. “gpy: a gaussian process framework in python.” http://github.com/\n",
      "sheffieldml/gpy.\n",
      "\u001b[0m\n",
      "\u001b[31mjones, donald r., matthias schonlau, and william j. welch. 1998. “efficient global\n",
      "optimization of expensive black-box functions.” journal of global optimization 13 (4):\n",
      "455–92. doi:10.1023/a:1008306431147.\n",
      "\u001b[0m\n",
      "\u001b[31mmcgibbon, robert t., matthew harrigan, bharath ramsundar, kyle beauchamp, mo-\n",
      "hammad m. sultan, christian schwantes, carlos x. hernández, et al. 2016. “msmbuilder:\n",
      "msmbuilder 3.4.” zenodo. doi:10.5281/zenodo.48545.\n",
      "\u001b[0m\n",
      "\u001b[31mmcgibbon, robert t., carlos x. hernández, matthew p. harrigan, steven kearnes, mo-\n",
      "hammad m. sultan, stanislaw jastrzebski, brooke e. husic, and vijay pande. 2016.\n",
      "“osprey 1.0.0.” doi:10.5281/zenodo.56251.\n",
      "\u001b[0m\n",
      "\u001b[31mpedregosa, f., g. varoquaux, a. gramfort, v. michel, b. thirion, o. grisel, m. blondel,\n",
      "\u001b[0m\n",
      "\u001b[31mmcgibbon et al., (2016). osprey: hyperparameter optimization for machine learning. journal of open source software, 1(5), 34,\n",
      "doi:10.21105/joss.00034\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n",
      "\u001b[31met al. 2011. “scikit-learn: machine learning in python.” journal of machine learning\n",
      "research 12: 2825–30.\n",
      "\u001b[0m\n",
      "\u001b[31myamins, daniel, david tax, and james s. bergstra. 2013. “making a science of model\n",
      "search: hyperparameter optimization in hundreds of dimensions for vision architec-\n",
      "tures.” in proceedings of the 30th international conference on machine learning (icml-\n",
      "13), edited by sanjoy dasgupta and david mcallester, 28:115–23. 1. jmlr workshop;\n",
      "conference proceedings. http://jmlr.csail.mit.edu/proceedings/papers/v28/bergstra13.\n",
      "pdf.\n",
      "\u001b[0m\n",
      "\u001b[31mmcgibbon et al., (2016). osprey: hyperparameter optimization for machine learning. journal of open source software, 1(5), 34,\n",
      "doi:10.21105/joss.00034\n",
      "\u001b[0m\n",
      "\u001b[31m2\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00035/10.21105.joss.00035.pdf\n",
      "\u001b[31mdoi: 10.21105/joss.00035\n",
      "\u001b[0m\n",
      "\u001b[31m1 aalto university\n",
      "\u001b[0m\n",
      "\u001b[31meetu mäkelä1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mlas: an integrated language analysis tool for multiple\n",
      "languages\n",
      "\u001b[0m\n",
      "\u001b[32mlas is a command-line tool for lemmatizing, morphological analysis, inflected form gen-\n",
      "eration, hyphenation and language identification of multiple languages.\n",
      "\u001b[0m\n",
      "\u001b[32mthese functionalities are of use as part of many workflows requiring natural language\n",
      "processing. indeed, las has been used for example as part of a pipeline for entity recog-\n",
      "nition (mäkelä 2014), in creating a contextual reader for texts in english, finnish and\n",
      "latin (mäkelä, lindquist, and hyvönen 2016), and for processing a finnish historical\n",
      "newspaper collection in preparation for data publication (pääkkönen et al. 2016).\n",
      "\u001b[0m\n",
      "\u001b[31mthe functionalities of las are mostly based on integrating existing tools into a common\n",
      "package. particularly, the tool bases on: * finite state transducers provided by the\n",
      "hfst (lindén et al. 2013), omorfi (t. a. pirinen 2015) and giellatekno (moshagen\n",
      "et al., n.d.) projects * snowball stemmers * the language-detector library * statistical\n",
      "language models from turku nlp (haverinen et al. 2014)\n",
      "\u001b[0m\n",
      "\u001b[32mwhile las supports many languages, the most complete support it has is for finnish,\n",
      "where considerable work has gone into improving the results.\n",
      "\u001b[0m\n",
      "\u001b[32maside from a being available as a command-line tool, the functionalities in las are also\n",
      "available as a web service, at http://demo.seco.tkk.fi/las/.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mhaverinen, katri, jenna nyblom, timo viljanen, veronika laippala, samuel kohonen,\n",
      "anna missilä, stina ojala, tapio salakoski, and filip ginter. 2014. “building the essen-\n",
      "tial resources for finnish: the turku dependency treebank.” language resources and\n",
      "evaluation 48 (3). springer netherlands: 493–531. doi:10.1007/s10579-013-9244-1.\n",
      "\u001b[0m\n",
      "\u001b[31mlindén, krister, erik axelson, senka drobac, sam hardwick, juha kuokkala, jyrki niemi,\n",
      "tommi pirinen, and miikka silfverberg. 2013. “hfst – a system for creating nlp tools.”\n",
      "in systems and frameworks for computational morphology, edited by cerstin mahlow and\n",
      "michael piotrowski, 380:53–71. communications in computer and information science.\n",
      "berlin heidelberg: springer.\n",
      "\u001b[0m\n",
      "\u001b[31mmäkelä, eetu. 2014. “combining a rest lexical analysis web service with sparql for\n",
      "mashup semantic annotation from text.” in the semantic web: eswc 2014 satel-\n",
      "lite events: eswc 2014 satellite events, anissaras, crete, greece, may 25-29, 2014,\n",
      "revised selected papers, edited by valentina presutti, eva blomqvist, raphael troncy,\n",
      "harald sack, ioannis papadakis, and anna tordai, 424–28. cham: springer international\n",
      "publishing. doi:10.1007/978-3-319-11955-7_60.\n",
      "\u001b[0m\n",
      "\u001b[31mmäkelä, eetu, thea lindquist, and eero hyvönen. 2016. “core - a contextual reader\n",
      "\u001b[0m\n",
      "\u001b[31mmäkelä, (2016). las: an integrated language analysis tool for multiple languages. journal of open source software, 1(6), 35,\n",
      "doi:10.21105/joss.00035\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n",
      "\u001b[32mbased on linked data.” in proceedings of digital humanities 2016, long papers. kraków,\n",
      "poland.\n",
      "\u001b[0m\n",
      "\u001b[31mmoshagen, sjur, jack rueter, tommi pirinen, trond trosterud, and francis m tyers. n.d.\n",
      "“open-source infrastructures for collaborative work on under-resourced languages.” in\n",
      "proceedings of the ninth international conference on language resources and evaluation,\n",
      "lrec, 71–77.\n",
      "\u001b[0m\n",
      "\u001b[31mpääkkönen, tuula, jukka kervinen, asko nivala, kimmo kettunen, and eetu mäkelä.\n",
      "2016. “exporting finnish digitized historical newspaper contents for offline use.” d-lib\n",
      "magazine 22 (7/8). corporation for national research initiatives. doi:10.1045/july2016-\n",
      "paakkonen.\n",
      "\u001b[0m\n",
      "\u001b[31mpirinen, tommi a. 2015.\n",
      "“omorfi—free and open source morphological lexical\n",
      "database for finnish.” in proceedings of the 20th nordic conference of computational\n",
      "linguistics, nodalida 2015, may 11-13, 2015, vilnius, lithuania, 313–15. 109. linköping\n",
      "university electronic press.\n",
      "\u001b[0m\n",
      "\u001b[31mmäkelä, (2016). las: an integrated language analysis tool for multiple languages. journal of open source software, 1(6), 35,\n",
      "doi:10.21105/joss.00035\n",
      "\u001b[0m\n",
      "\u001b[31m2\n",
      "\u001b[0m\n",
      "/Volumes/Seagate Backup Plus Drive/JOSS Project/joss-papers-master/joss-papers-master/joss.00036/10.21105.joss.00036.pdf\n",
      "\u001b[32mgwdegree: improving interpretation of\n",
      "geometrically-weighted degree estimates in exponential\n",
      "random graph models\n",
      "\u001b[0m\n",
      "\u001b[31mdoi: 10.21105/joss.00036\n",
      "\u001b[0m\n",
      "\u001b[31m1 university of california, davis; department of environmental science and policy\n",
      "\u001b[0m\n",
      "\u001b[31mmichael a levy1\n",
      "\u001b[0m\n",
      "\u001b[31msummary\n",
      "\u001b[0m\n",
      "\u001b[31msoftware\n",
      "\u001b[0m\n",
      "\u001b[31m• review\n",
      "• repository\n",
      "• archive\n",
      "\u001b[0m\n",
      "\u001b[31mlicence\n",
      "authors of joss papers retain\n",
      "copyright and release the work un-\n",
      "der a creative commons attri-\n",
      "bution 4.0 international license\n",
      "(cc-by).\n",
      "\u001b[0m\n",
      "\u001b[32mexponential random graph models (ergms) are maximum entropy statistical models\n",
      "that provide estimates on network tie formation of variables both exogenous (covariate)\n",
      "and endogenous (structural) to a network. network centralization – the tendency for\n",
      "edges to accrue among a small number of popular nodes – is a key network variable in\n",
      "many fields, and in ergms it is primarily modeled via the geometrically-weighted degree\n",
      "(gwd) statistic (snijders et al. 2006; hunter 2007). however, the published literature is\n",
      "ambiguous about how to interpret gwd estimates, and there is little guidance on how\n",
      "to interpret or fix values of the gwd shape-parameter, (cid:18)s. this shiny application seeks\n",
      "to improve the use of gwd in ergms by demonstrating:\n",
      "\u001b[0m\n",
      "\u001b[32m1. how the gwd statistic responds to adding edges to nodes of various degrees, con-\n",
      "\u001b[0m\n",
      "\u001b[32mtingent on the value of the shape parameter, (cid:18)s;\n",
      "\u001b[0m\n",
      "\u001b[32m2. how the degree distribution of networks of various size and density are shaped by\n",
      "\u001b[0m\n",
      "\u001b[32mgwd parameter and (cid:18)s values;\n",
      "\u001b[0m\n",
      "\u001b[32m3. how gwd and gwesp – an ergm term used to model triadic closure – interact\n",
      "\u001b[0m\n",
      "\u001b[32mto affect network centralization and clustering.\n",
      "\u001b[0m\n",
      "\u001b[31mreferences\n",
      "\u001b[0m\n",
      "\u001b[31mhunter, david r. 2007. “curved exponential family models for social networks.” social\n",
      "networks, special section: advances in exponential random graph (p*) models, 29 (2):\n",
      "216–30. doi:10.1016/j.socnet.2006.08.005.\n",
      "\u001b[0m\n",
      "\u001b[31msnijders, tom a. b., philippa e. pattison, garry l. robins, and mark s. handcock. 2006.\n",
      "“new specifications for exponential random graph models.” sociological methodology 36\n",
      "(1): 99–153. doi:10.1111/j.1467-9531.2006.00176.x.\n",
      "\u001b[0m\n",
      "\u001b[31mlevy, (2016). gwdegree: improving interpretation of geometrically-weighted degree estimates in exponential random graph models. journal of\n",
      "open source software, 1(3), 36, doi:10.21105/joss.00036\n",
      "\u001b[0m\n",
      "\u001b[31m1\n",
      "\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "for K in range(20):\n",
    "    print(PAPER_OF_INTEREST_FNAME[K])\n",
    "    Paper_interest = PAPER_OF_INTEREST_FNAME[K]\n",
    "    Show_Cleaned_text(Paper_interest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d47a43b4-0fc5-4810-859d-b05973aa7547",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f43a9dca-d24b-434f-b2bd-3f6dfa1c8e41",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e5e54b33-674e-4e49-8b48-7e5037ab5b5b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df9ffc63-7964-4fce-8bef-da9922e3f37c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f7147a1a-ac93-4217-aced-d761707d01e5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ae68417-5fed-469d-9190-69630819d453",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d4689a4f-33bb-44a6-b1a1-babdb94bc945",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0608f93d-d2d8-45a3-9d65-ce630868a800",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd587b65-9923-425e-b055-177722f3ebd8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "490b73dd-93d2-4aa4-a599-482295ad96ca",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4acdf8dd-a1a4-41cb-a7b6-138ceca38edd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40961b55-b593-4c2d-87cd-dc78090cb6b3",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
